lrtable/src/lib/statetable.rs

```diff
@@ -303,10 +303,8 @@ where
     }
 
     fn decode(bits: u32) -> Action<StorageT> {
-        //           Action   State/Prod
-        let mask = 0b00000000_11111111;
-        let action = bits >> 8;
-        let reference = bits & mask;
+        let action = bits & 0b11;
+        let reference = bits >> 2;
 
         if action == 1 {
```

Comment:
I suggest you make an `enum` for these. It'll be something like:

```
enum EncodedActions { ACCEPT = 0; REDUCE = 1; ... }

...

match action {
  ACCEPT => Action::Accept,
  REDUCE => Action::Reduce(val)
  ...
}
```

then you can statically prove that the `else` branch can't happen (so there's no need to dynamically panic).

Reply:
I was about to ask how I can make this more elegant! ;-)